packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -0,0 +1,146 @@
+        vscode.window.onDidChangeActiveTextEditor((message: any) => {
+            const editedFile = vscode.window.activeTextEditor?.document
+            this.onChangeInputPath.fire(editedFile!.uri.path)
+        })
+    }
+    public _setActiveConfigurationListener() {
+        vscode.workspace.onDidChangeConfiguration((config: vscode.ConfigurationChangeEvent) => {
+            // If settings change, we want to update the Webview to reflect the change in inputs
+            if (config.affectsConfiguration('aws.accessAnalyzer.policyChecks.referencePolicyFilePath')) {
+                this.onChangeReferenceFilePath.fire(
+                    vscode.workspace.getConfiguration().get('aws.accessAnalyzer.policyChecks.referencePolicyFilePath')!
+                )
+            } else if (config.affectsConfiguration('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')) {
+                this.onChangeCloudformationParameterFilePath.fire(
+                    vscode.workspace
+                        .getConfiguration()
+                        .get('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')!
+                )
+            }
+        })
+    }
+
+    // TODO: Add functionality for the three checks
+    public async validatePolicy(documentType: string, policyType: string) {
+        this.client /* Do something */
+    }
+    public async checkNoNewAccess() {}
+    public async checkAccessNotGranted() {}
+}
+
+const Panel = VueWebview.compilePanel(IamPolicyChecksWebview)
+
+export async function renderIamPolicyChecks(context: ExtContext): Promise<void> {
+    const logger: Logger = getLogger()
+    try {
+        const client = new AccessAnalyzer({ region: context.regionProvider.defaultRegionId })
+        //Read from settings to auto-fill some inputs
+        const referencePolicyFilePath: string = vscode.workspace
+            .getConfiguration()
+            .get('aws.accessAnalyzer.policyChecks.referencePolicyFilePath')!
+        const cfnParameterPath: string = vscode.workspace
+            .getConfiguration()
+            .get('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')!
+
+        //Check if Cfn and Tf tools are installed
+        var cfnToolInstalled = true
+        var tfToolInstalled = true
+        exec('tf-policy-validator', err => {
+            if (err) {
+                if (err.message.includes('command not found')) {
+                    tfToolInstalled = false
+                }
+            }
+        })
+        exec('cfn-policy-validator', err => {
+            if (err) {
+                if (err.message.includes('command not found')) {
+                    cfnToolInstalled = false
+                }
+            }
+        })
+
+        const wv = new Panel(
+            context.extensionContext,
+            {
+                referenceFilePath: referencePolicyFilePath ? referencePolicyFilePath : '',
```

Comment:
Q: Isn't this a required parameter?

Reply:
Yes, but it is not always provided at the start. It is inputted by the user through a setting or by inputting it directly in the Webview.